### PR TITLE
Add Manwa

### LINKED ISSUE #14234: manwa.me
### Source name

Manwa

### Source link

https://manwa.me/

### Language

Chinese

### Other details

18+/NSFW = yes
hentai, pornhwa, 18+ sources with some non-18+ source.
Previously requested but closed without being addressed. 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by anenasa**
Closes #14234 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/zh/manwa/src/eu/kanade/tachiyomi/extension/zh/manwa/Manwa.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    override fun headersBuilder(): Headers.Builder = Headers.Builder().add("User-Agent", USER_AGENT)
+
+    private val rewriteOctetStream: Interceptor = Interceptor { chain ->
+        val originalResponse: Response = chain.proceed(chain.request())
+        if (originalResponse.request.url.toString().endsWith("?v=20220724")) {
+            // Decrypt images in mangas
+            val orgBody = originalResponse.body!!.bytes()
+            val key = "my2ecret782ecret".toByteArray()
+            val aesKey = SecretKeySpec(key, "AES")
+            val cipher = Cipher.getInstance("AES/CBC/NOPADDING")
```
Comment (AntsyLich) [inline src/zh/manwa/src/eu/kanade/tachiyomi/extension/zh/manwa/Manwa.kt:56]: Use the aes decryption library in `/lib`

Comment (stevenyomi) [reply src/zh/manwa/src/eu/kanade/tachiyomi/extension/zh/manwa/Manwa.kt:56]: > Use the aes decryption library in `/lib`

No, that library only helps decoding specific types of cipher & password and won't work here.

Comment (anenasa) [reply src/zh/manwa/src/eu/kanade/tachiyomi/extension/zh/manwa/Manwa.kt:55]: That library is not working for me, I believe the [toString() call](https://github.com/tachiyomiorg/tachiyomi-extensions/blob/332e5950d073d8b31027e303fb710e4b1b39221c/lib/cryptoaes/src/main/java/eu/kanade/tachiyomi/lib/cryptoaes/CryptoAES.kt#L74) destroys the decryption result.

File: src/zh/manwa/src/eu/kanade/tachiyomi/extension/zh/manwa/Manwa.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            key = IMAGE_HOST_KEY
+            title = "图源"
+            entries = IMAGE_HOST_ENTRIES
+            entryValues = IMAGE_HOST_ENTRY_VALUES
+            setDefaultValue(IMAGE_HOST_ENTRY_VALUES[0])
+        }.let { screen.addPreference(it) }
+    }
+
+    companion object {
+        private const val USER_AGENT = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"
```
Comment (alessandrojean) [inline src/zh/manwa/src/eu/kanade/tachiyomi/extension/zh/manwa/Manwa.kt:170]: Is this really really necessary? If it's not the case, it is better to not set so it reuses the app one which the user can change if needed.

**Pull request merged**